src/net/KNet/Specific/Serialization/KNetSerialization.cs

```diff
@@ -290,7 +290,7 @@ public static float DeserializeFloat(string topic, byte[] data)
         /// </summary>
         public static int DeserializeInt(string topic, byte[] data)
         {
-            return (int)_IntDeserializer.Deserialize(topic, data);
+            return JVMBridgeBase.Wraps<Java.Lang.Integer>(_IntDeserializer.Deserialize(topic, data) as IJavaObject);
```

Comment:
Running the tests related to `KNetCompactedReplicator` seems the result of `Deserialize` is not always a Java object. I don't understand why but the result is an `int` instead of `java.lang.Integer` even if the Java signature is `public Integer deserialize(String topic, byte[] data)`. Maybe a better implementation needs a check against `IJavaObject`.